### PR TITLE
(947) Allow the index offence to be selected at the start of the journey if an offender has more than one offence

### PR DESCRIPTION
**Pull request by pezholio**
This adds a new page to the start of the Apply journey if an offender has more than one offence associated with them. The guidance notes could use some work, but this is probably enough as a started for ten.

## Screenshot

image

### CODE REVIEW

**Comment by pezholio** [comment]
[Allow the index offence to be selected at the start of the journey if an offender has more than one index offence](https://trello.com/c/UQHa6V4B/947-allow-the-index-offence-to-be-selected-at-the-start-of-the-journey-if-an-offender-has-more-than-one-index-offence)

File: server/controllers/apply/applicationsController.ts [hunk 1]
```diff
@@ -51,17 +51,14 @@ export default class ApplicationsController {
         const person = await this.personService.findByCrn(req.user.token, crn)
         const offences = await this.personService.getOffences(req.user.token, crn)
 
-        // TODO: For now, we treat the first offence as the index offence, going
-        // forward, we'll need to design an approach to select one if there are
-        // more than one
-        const offence = offences[0]
+        const offenceId = offences.length === 1 ? offences[0].offenceId : undefined
```
Comment (richpjames) [inline server/controllers/apply/applicationsController.ts:54]: I don't feel too strongly about it but if we're setting a variable to be falsy I prefer `null` over `undefined` as it makes things easier to debug if something goes wrong.

File: server/views/applications/people/selectOffence.njk [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-full">
+      <h1>{{ pageHeading}}</h1>
+
+      <p class="govuk-body">
+        {{ person.name }} has more than one offence identified against their CRN.
+
+      <p class="govuk-body">
+        Please select the main offence that will be used to assess their sutability
+        for an Approved Premises (the "index offence") below:
```
Comment (richpjames) [inline server/views/applications/people/selectOffence.njk:20]: I think this should be `'` as opposed to `"` but, as ever, happy to be wrong!

**Review by richpjames — APPROVED** [review]
LGTM

File: server/controllers/apply/applicationsController.ts (hunk 1 above)
Comment (pezholio) [reply server/controllers/apply/applicationsController.ts]: Yeah, good call 👍

**Pull request merged**